core/src/main/java/org/ehcache/core/Ehcache.java

```diff
@@ -648,6 +648,69 @@ public void loadAll(Set<? extends K> keys, boolean replaceExistingValues, Functi
       }
     }
 
+    /**
+     * The reason behind this method's name is because the 107 standard is so fucked up in so many mixed aspects
+     * <b>and</b> its TCK is so ridiculously relying on implementation details when it actually does test something
+     * remotely useful, Ehcache's 'sane' iterator contract (whatever sane might mean in this context) makes
+     * the 107 TCK unhappy.
+     *
+     * The 107 iteration contract combined with stats and expiration is so mind-blowingly lame that there is no way
+     * to adapt Ehcache's iterator. A secondary, really, really crappy one must be implemented.
+     */
+    @Override
+    public Iterator<Entry<K, V>> crappyterator() {
+      final Store.Iterator<Entry<K, ValueHolder<V>>> iterator = store.iterator();
+      return new Iterator<Entry<K, V>>() {
+        K current;
+        @Override
+        public boolean hasNext() {
+          return iterator.hasNext();
+        }
+
+        /**
+         * The multiple get calls are a necessary stupidity to please the 107 TCK's stats and
+         * you-cannot-expire-on-creation-but-immediately-expire-upon-retrieval-but-not-the-first-one-of-course contract.
+         */
+        @Override
+        public Entry<K, V> next() {
+          try {
+            final Entry<K, ValueHolder<V>> next = iterator.next();
+            // call Cache.get() here to check for expiry *and* account for a get in the stats
+            if (get(next.getKey()) == null) {
+              current = null;
+              return null;
+            }
+            // call Store.get() here to check check for hasNext()' expiry *and do not* account for an extra get in the stats
+            store.get(next.getKey());
```

Comment:
without the call to Store.get(), CacheExpiryTest.expire_whenAccessed() assertion at line 371 fails because iterator.hasNext() returns true while the store only contains expired mappings.